packages/ckeditor5-widget/src/widget.js

```diff
@@ -126,21 +126,33 @@ export default class Widget extends Plugin {
 
 			let lastMarked = null;
 
+			let rangeCounter = 0;
+			const widgetsToSelect = [];
+
 			for ( const range of viewSelection.getRanges() ) {
+				// if there is more than 1 range it means that we are inside table and we don't want to select nested widgets.
```

Comment:
If really the only correct behaviour is when we do have only one range, and we ignore other ones, then we do not need to loop through all of them at all. We can just pick the first range from `viewSelection.getRanges()`.

In such case there is no need for `rangeCounter`, widgets accumulation and later for the additional loop.

Reply:
Also, I would think about it twice. Is there really no other case where there are multiple ranges but we are not inside table? In other words is there a situation where multiple ranges are expected and we need to handle all of them?

Reply:
This change is reverted.